File: Java/Ficheros/ficheros/ejercicio6/Ejercicio6.java
package ejercicio6;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

/**
 * Realiza un programa que diga cuántas ocurrencias de una palabra hay en un
 * fichero. Tanto el nombre del fichero como la palabra se deben pasar como
 * argumentos en la línea de comandos.
 * 
 * @author Juan Manuel Gonzalez Profumo
 * @version 1.0
 */
public class Ejercicio6 {
	public static void main(String[] args) {
		
		try {
			BufferedReader br = new BufferedReader(new FileReader(args[0] + ".txt"));
			ArrayList<String> palabras = new ArrayList<String>();
			String palabra = "" + args[1];
			int repetida = 0;
			
			String linea = "";
			
			while (linea != null) {
				palabras.add(linea);
				linea = br.readLine();
			}
						
			for (String s : palabras) {
				if (s.equals(palabra)) {
					repetida++;
				}
			}
			
			br.close();
			
			System.out.println("La palabra " + palabra + " se repite en el fichero " + repetida + " veces");
			
		} catch (FileNotFoundException e) {
			System.err.println("No se encuentra el fichero");
		} catch (IOException e) {
			System.err.println("No se puede leer el fichero");
		}
		
	}

}

File: Python/Ficheros/ejercicio6/Ejercicio6.py
nombreArchivo = input("Introduce el nombre del fichero: ")
archivo = open(nombreArchivo + '.txt', 'r')
lista = []
palabra  = input("Introduce la palabra que quiera buscar: ")
repetida = 0

while True:
    linea = archivo.readline()
    if not linea:
        break
    lista.append(linea.strip())

for s in lista:
    if s == palabra:
        repetida = repetida + 1

archivo.close()
print("La palabra " + palabra + " se repite en el fichero " + str(repetida) + " veces")

